### PR TITLE
Allow limited multiple dereferences

### PR DESCRIPTION
**Pull request by scarabus**
if "FOO" = "$BAR" 
and TODAY = "#010101"
and BAR = "%!document://FOO/${TODAY}#foo"
then FOO = {
 x <-- document://FOO/010101
 return x.foo
}

### CODE REVIEW

**Comment by scarabus** [comment]
Oops - didn't mean to bump the version

File: Libs/RaptureCore/src/main/java/rapture/kernel/dp/ExecutionContextUtil.java [hunk 1]
```diff
@@ -51,33 +51,44 @@ public static String getValueECF(CallingContext callingContext, String workOrder
...
-        if (valueType == ContextValueType.VAR) {
-            /*
-             * If a variable, first retrieve the val we need to evaluate
-             */
-            String idNoMarker = realId.substring(1);
-            ExecutionContextField ecf = ExecutionContextFieldStorage.readByFields(workOrderUri, idNoMarker);
-            if (ecf != null) {
-                toEval = ecf.getValue();
+
+        while (true) {
```
Comment (jonathan-major) [inline Libs/RaptureCore/src/main/java/rapture/kernel/dp/ExecutionContextUtil.java]: whats this for?

**Comment by scarabus** [comment]
To allow multiple dereferences.

**Comment by scarabus** [comment]
If a variable is say %!//${foo} then the % denotes a template. So that
dereferernces to !//foo which is a link so next time around the link is
resolved.

**Pull request merged**